tests/altair_theme_test.py

```diff
@@ -0,0 +1,276 @@
+      <option value="urbaninstitute">urbaninstitute</option>
+      <option value="googlecharts">googlecharts</option>
+      <option value="powerbi">powerbi</option>
+      <optgroup label="Carbon">
+        <option value="carbonwhite">carbonwhite</option>
+        <option value="carbong10">carbong10</option>
+        <option value="carbong90">carbong90</option>
+        <option value="carbong100">carbong100</option>
+      </optgroup>
+    </select>
+    <br />
+  </div>
+  <div id="{{ output_div }}"></div>
+  <script>
+    // Vega Theme Test - But single view, no renderer option
+    var spec = {{ spec }};
+    var container = document.querySelector("#{{ output_div }}");
+    var themes = document.querySelector("#themes");
+    var currentLocation = window.location;
+    var url = new URL(currentLocation);
+
+    themes.addEventListener("change", function () {
+      theme = themes.options[themes.selectedIndex].value;
+      url.searchParams.set("theme", `${themes.options[themes.selectedIndex].value}`);
+      window.history.replaceState(null, null, url);
+      refresh();
+    });
+
+    function refresh() {
+      const themeName = themes.options[themes.selectedIndex].value;
+      container.innerHTML = "";
+      var el = document.createElement("div");
+      el.setAttribute("class", "view");
+      container.appendChild(el);
+      vegaEmbed(el, spec, {
+          theme: themeName,
+          defaultStyle: true,
+          mode: "vega-lite"
+      });
+    }
+
+    // Mostly the standard template
+    // - Need to adapt these two ideas into one
+    (function(vegaEmbed) {
+      var embedOpt = { mode: "vega-lite" };
+
+      function showError(el, error){
+          el.innerHTML = ('<div style="color:red;">'
+                          + '<p>JavaScript Error: ' + error.message + '</p>'
+                          + "<p>This usually means there's a typo in your chart specification. "
+                          + "See the javascript console for the full traceback.</p>"
+                          + '</div>');
+          throw error;
+      }
+      const el = document.getElementById('{{ output_div }}');
+      vegaEmbed("#{{ output_div }}", spec, embedOpt)
+        .catch(error => showError(el, error));
+    })(vegaEmbed);
+
+  </script>
+</body>
+</html>
+"""
+)
+
+
+def render_write(fp: str | Path, /) -> None:
+    """
+    Debug tool, effectively `alt.Chart.save` w/ template.
+
+    ## Remove before review
+    """
+    content = TEMPLATE.render(
+        spec=json.dumps(alt_theme_test().to_dict(context={"pre_transform": False})),
+        vegalite_version=alt.VEGALITE_VERSION,
+        vegaembed_version=alt.VEGAEMBED_VERSION,
+        vega_version=alt.VEGA_VERSION,
+        base_url="https://cdn.jsdelivr.net/npm",
+        output_div="vis",
+    )
+    if isinstance(fp, (str, Path)):
+        with Path(fp).open(mode="w", encoding="utf-8") as f:
+            f.write(content)
+    else:
+        raise TypeError(fp)
```

Comment:
### TODO
- Make this `main()`
- Accept `fp` via command/argparse
- Move module to `tools/...`